File: api/Models/Topic.cs
using System.ComponentModel.DataAnnotations.Schema;
namespace api.Models
{
    [Table("Tema")]
    public class Topic
    {
        [Column("id_teme")]
        public int Id { get; set; }
        [Column("naziv")]
        public string Title { get; set; } = null!;
        [Column("opis")]
        public string Description { get; set; } = null!;
        [Column("status")]
        public int Status { get; set; }
        [Column("id_predmeta")]
        public int subjectId { get; set; }
        [Column("id_profesora")]
        public int professorId { get; set; }
        [Column("id_ucenika")]
        public int? studentId { get; set; }
        public User User { get; set; } = null!;
        public Subject Subject { get; set; } = null!;
        public Student? Student { get; set; }
    }
}

File: api/app/Models/Topic.php
<?php

namespace App\Models;

use Illuminate\Database\Eloquent\Factories\HasFactory;
use Illuminate\Database\Eloquent\Model;
use Illuminate\Database\Eloquent\Relations\BelongsTo;

class Topic extends Model
{
    use HasFactory;
    
    const TABLE = 'topics';
    protected $table = self::TABLE;
    
    protected $fillable = [
        'title',
        'description',
        'status',
        'professor_subject_id',
        'student_id'
    ];

    public function professor_subject(): BelongsTo
    {
        return $this->belongsTo(ProfessorSubject::class);
    }

    public function student(): BelongsTo
    {
        return $this->belongsTo(Student::class);
    }
}